### PR TITLE
S3878: Add FP reproducers for #6893 and #6894

### PR DESCRIPTION
**Pull request by mary-georgiou-sonarsource**
Repros for #6894 #6893

### CODE REVIEW

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/ArrayPassedAsParams.cs [hunk 1]
```diff
@@ -71,3 +71,25 @@ public class IndexerClass
...
+public class Repro6894
+{
+    //Reproducer for https://github.com/SonarSource/sonar-dotnet/issues/6894
+
+    public void Method(params object[] args) { }
+    public void MethodJuggedArray(params int[][] args) { }
+
+    public void CallMethod()
+    {
+        Method(new String[] { "1", "2" }); // Noncompliant, TP. Elements in args: {"1", "2"}
```
Comment (martin-strecker-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/TestCases/ArrayPassedAsParams.cs:84]:
```suggestion
        Method(new String[] { "1", "2" }); // Noncompliant, FP. Elements in args: {string[]}
```

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/ArrayPassedAsParams.cs [hunk 2]
```diff
@@ -71,3 +71,25 @@ public class IndexerClass
 {
     public int this[params int[] i] => 1;
 }
+
+public class Repro6894
+{
+    //Reproducer for https://github.com/SonarSource/sonar-dotnet/issues/6894
+
+    public void Method(params object[] args) { }
```
Comment (martin-strecker-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/TestCases/ArrayPassedAsParams.cs:79]: I think `params Array[]` causes also FPs. Can you add some test cases?
```suggestion
    public void Method(params object[] args) { }
    public void MethodArray(params Array[] args) { }
```

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/ArrayPassedAsParams.cs [hunk 3]
```diff
@@ -71,3 +71,25 @@ public class IndexerClass
...
+    public void MethodJuggedArray(params int[][] args) { }
+
+    public void CallMethod()
+    {
+        Method(new String[] { "1", "2" }); // Noncompliant, TP. Elements in args: {"1", "2"}
+        Method(new object[] { new int[] { 1, 2} }); // Noncompliant. Elements in args: {System.Int32[]}
```
Comment (martin-strecker-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/TestCases/ArrayPassedAsParams.cs:85]:
```suggestion
        Method(new object[] { new int[] { 1, 2} }); // Noncompliant, TP. Elements in args: {System.Int32[]}
```

**Review by martin-strecker-sonarsource — CHANGES_REQUESTED** [review]

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/ArrayPassedAsParams.cs (hunk 1 above)
Comment (mary-georgiou-sonarsource) [reply analyzers/tests/SonarAnalyzer.UnitTest/TestCases/ArrayPassedAsParams.cs:84]: @martin-strecker-sonarsource  it's not the case :/ For me when I iterate over args it's giving me back two elements 1 and 2.
Could you maybe run a quick test on your side too?

Comment (mary-georgiou-sonarsource) [reply analyzers/tests/SonarAnalyzer.UnitTest/TestCases/ArrayPassedAsParams.cs:84]: I added a comment there to clarify.

**Comment by mary-georgiou-sonarsource** [comment]
@martin-strecker-sonarsource  I also added a repro for #6893 since it's small and related to the same rule.

**Pull request merged**